From 613571635409f08c9eb0460e011616dd16250ecb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=85ke=20Rehnman?= <ake_rehnman_at_gmail_com>
Date: Sat, 4 Mar 2017 00:06:10 +0100
Subject: [PATCH 2/10] added clear_proceed_status in run_command_1 causing run
 command to hang.

---
 gdb/infcmd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gdb/infcmd.c b/gdb/infcmd.c
index 976276b694..2050634c29 100644
--- a/gdb/infcmd.c
+++ b/gdb/infcmd.c
@@ -675,6 +675,7 @@ run_command_1 (const char *args, int from_tty, enum run_how run_how)
 
   /* Start the target running.  Do not use -1 continuation as it would skip
      breakpoint right at the entry point.  */
+  clear_proceed_status (0);
   proceed (regcache_read_pc (get_current_regcache ()), GDB_SIGNAL_0);
 
   /* Since there was no error, there's no need to finish the thread
-- 
2.12.2
